pycheribuild/projects/cross/opensbi.py

```diff
@@ -201,3 +217,42 @@ def run_tests(self):
             give_tty_control=True,
             cwd="/",
         )
+
+
+class BuildAllianceOpenSBI(BuildOpenSBI):
+    target = "cheri-alliance-opensbi"
+    _default_install_dir_fn = ComputedDefaultValue(
+        function=lambda config, p: config.cheri_alliance_sdk_dir / "opensbi/riscv64",
+        as_string="$SDK_ROOT/opensbi/riscv64",
+    )
+    repository = GitRepository("https://github.com/CHERI-Alliance/opensbi", default_branch="codasip-cheri-riscv")
+    supported_architectures = (
+        CompilationTargets.FREESTANDING_RISCV64,
+        CompilationTargets.FREESTANDING_RISCV64_PURECAP,
+    )
+
+    def _qemu_install_dir(self) -> Path:
+        return BuildCheriAllianceQEMU.get_install_dir(self, cross_target=CompilationTargets.NATIVE)
+
+    def setup(self):
+        super().setup()
+        self.make_args.set(FW_TEXT_START=0x80000000)
+
+    @property
+    def all_platforms(self):
+        return ["generic"]
+
+    def compile(self, **kwargs):
+        for platform in self.all_platforms:
+            args = self.make_args.copy()
+            args.set(PLATFORM=platform)
+            self.run_make(parallel=False, cwd=self.source_dir, options=args)
+
+
+class BuildAllianceOpenSBIGFE(BuildAllianceOpenSBI):
+    target = "cheri-alliance-opensbi-gfe"
+    repository = ReuseOtherProjectRepository(BuildAllianceOpenSBI, do_update=True)
+
+    def setup(self):
+        super().setup()
+        self.make_args.set(FW_TEXT_START=0xC0000000)
```

Comment:
If the code were ported to self-relocate we wouldn't need a separate GFE one, we could just load it to the different address...